### DISCUSSION TITLE
The reason why builtin osc can be disabled by scripts

### DISCUSSION THREAD

**Discussion by po5** [tomasklaen/uosc#736] [General]
Think I've finally figured it out (:  
Turns out it's not a weird racy quirk where the property needs to be set to `no` quickly, it's actually an intended feature and works anytime. Can't believe I've never tried `set osc no` at runtime before.

More info: https://github.com/mpv-player/mpv/issues/12475#issuecomment-1778321830

Hope more UI authors learn about this to streamline install.

**Comment by hooke007** [comment]
https://github.com/po5/thumbfast/issues/66#issuecomment-1528869629
But some like renaming it with the same name of built-in osc.

https://github.com/tomasklaen/uosc/pull/128#issuecomment-1226435297
And later I did meet with this case occasionally.

**Reply by po5** [reply]
If the user is renaming your custom UI they don't get to complain that their keybinds or whatever don't magically update, that's a non-issue.

Flickering can only happen if you have it in an observe_property like uosc does. If you set the property at the top of your script it will happen during initialization before even the first event loop which is blocking and the player can't even render before then.

**Comment by po5** [comment]
Found a way to reclaim the original script name https://github.com/po5/thumbfast/commit/04819fc9ce49b67dfc2bac71c26e037bcb57b9d4 @hooke007
